### ISSUE TITLE
[iOS 13.x] Textfield focus after route navigate 

### ISSUE THREAD

**Issue by bbaki** [NativeScript/nativescript-angular#2096]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI:  `6.2.2` (`tns --version`)
 - iOS Runtime: `6.2.0`
 - NativeScript-Angular: `~8.20.0`
 - Angular: `~8.20.0`
 
**Describe the bug**
After updating an iPhone from iOS version 12.x to 13.x there is a textfield focus glitch after navigating to a certain page. The page opens once, then the keyboard appears and the page appears another time.
![](https://s5.gifyu.com/images/ezgif.com-video-to-gifa5c42453d0db6d79.gif)

For this example I use a simple `setTimeout` 100ms because without it the element won't be focus. In our real application we try to focus an element in an interval and stop when the element receives focus. 

The relevant code for our focus function:
Typescript:
```
export class ItemDetailComponent implements AfterViewInit {
    @ViewChild('textField', { static: false }) textField: ElementRef

    ngAfterViewInit(): void {
        setTimeout(() => {
            this.textField.nativeElement.focus()
        }, 100)
    }
}
```

Html:
```
<FlexboxLayout flexDirection="column">
    <TextField #textField></TextField>
</FlexboxLayout>
```

**To Reproduce**
Navigate to a page and try to focus a textfield

**Expected behavior**
Focus should be done without any glitch or timeout needed :)

**Sample project**
See folder: `/nsiosfocus/src/app/item`

https://github.com/bbaki/nativescript-textfield-focus

**Additional context**
We get this issue on an iPhoneX with iOS 13.1.3. This issue seems to be reproducible both on a simulator and a real device. We did not have this problem on version 12.x nor does it occur on Android.

I'm not sure if the problem is with NS, routing or our code. I have tried to find a different solution for focusing but did not find one.

NS is awesome, keep up the good work :)

**Comment by lini** [comment]
I think this is related somehow to the page transition animation. If the timeout is shorter than the transition, the keyboard's own show animation interferes with the page transition. If you try a bigger value (e.g. 400) for the setTimeout/setInterval, it should work without glitches. Alternatively, you can experiment with the value of the page transition (https://docs.nativescript.org/api-reference/interfaces/_ui_frame_.navigationtransition#name) and keep the short timeout of 100ms, e.g.:

```HTML
<Label [nsRouterLink]="['/item', item.id]" pageTransition="fade" [text]="item.name"></Label>
```

**Comment by bbaki** [comment]
I would prefer to avoid setTimeouts since we get strange results with it. For example if I set the setTimeout to 500ms the keyboard appears after 2 seconds. This may be a problem with our code somewhere else but that still doesn't resolve the issue itself. Why did it work on iOS 12.X and not on 13.X? And how to solve this without such hax?

**Comment by edusperoni** [comment]
@bbaki You could use page events https://docs.nativescript.org/ui/components/page#page-events

```
    constructor(private page: Page) {
        page.on("navigatingTo", () => { console.log("navigatingTo", Profiling.time()); });
        page.on("navigatedTo", () => { console.log("navigatedTo", Profiling.time()); });
    }
```

It seems `navigatedTo` triggers after the page has fully transitioned, as can be seen in this demo:

https://play.nativescript.org/?template=play-ng&id=60QGDL

navigatedTo will trigger 1s after navigatingTo because I set a 1000ms transition.

**Comment by bbaki** [comment]
I don't think that the problem has been completely solved but we found a nice workaround. We have a directive that continuous to focus one input when the user is on a particular page. The initial focus is now done through `page.on("navigatedTo",` thanks to @edusperoni (we kind of had it already implemented but a fresh look allowed us to implement it in a different way). The initial focus works fine on both platforms. The problem start when we lose focus and want to autofocus that input again, luckily the keyboard we use on iOS does not seem to have an option to lose focus so we simply disable the autofocus option on iOS. For android we keep it on a 500ms loop and autofocus.

When writing this message I think I found a final solution. We should enable autofocus on iOS on `navigatedTo` and not ngOnInit. I did not test it though. I think we can close this issue.
